mausignald/signald.py

```diff
@@ -389,6 +389,29 @@ async def get_group(
             return None
         return GroupV2.deserialize(resp)
 
+    async def create_group(
+        self,
+        username: str,
+        avatar_path: str | None = None,
+        member_role_administrator: bool = False,
+        members: list[Address] | None = None,
+        title: str | None = None,
+    ) -> GroupV2 | None:
+        create_params = {
+            key: value
+            for key, value in {
+                "avatar": avatar_path,
+                "member_role": "ADMINISTRATOR" if member_role_administrator else "DEFAULT",
+                "title": title,
+                "members": [addr.serialize() for addr in members],
+            }.items()
+            if value is not None
+        }
```

Comment:
I think it's better to do the filter on a separate line.

```suggestion
        create_params = {
            "avatar": avatar_path,
            "member_role": "ADMINISTRATOR" if member_role_administrator else "DEFAULT",
            "title": title,
            "members": [addr.serialize() for addr in members],
        }
        create_params = {k: v for k, v in create_params.items() if v}
```

Reply:
I have no opinion on that, I just copied it from somewhere else. I suppose it was done like that for a reason?

Reply:
the "update_group" request follows the same pattern for its parameters. I suppose we should either change both or leave this one the way it is.

Reply:
I think the two-statement way is clearer what's going on. Let's change in both places.

Reply:
I figured out that these two statements are not equivalent. The version I originally used seems to create an empty members list if there are no members, the one you suggested entirely omits the members command. The former is permitted, the latter is not. Same for an empty title.

Reply:
The difference is `if v` vs `if value is not None`

Reply:
I see. I will adapt it accordingly. Being able to create empty groups and groups without a title should be possible if signal allows it.